src/login/logind-action.c

```diff
@@ -11,29 +13,119 @@
 #include "logind-dbus.h"
 #include "logind-session-dbus.h"
 #include "process-util.h"
-#include "sleep-config.h"
 #include "special.h"
 #include "string-table.h"
 #include "terminal-util.h"
 #include "user-util.h"
 
+static const ActionTableItem action_table[_HANDLE_ACTION_MAX] = {
+        [HANDLE_POWEROFF] = {
+                SPECIAL_POWEROFF_TARGET,
```

Comment:
we'd usually use C99 structured initialization for things like this, to make this less opaque and make it easy to refactor the structure one day. i.e. i'd prefer if this was written as:

```
…
        [HANDLE_POWEROFF] = {
                .target = …,
                .inhibit_what = …,
        }
…
```

Yes, this will be longer, but I think it still makes it more readable given how many fields this has (and how non-obvious they are). Moreover, we kinda do this in most other similar cases (though admittedly, not in all)

Reply:
ok, will do. Wasn't sure about that as the whole thing is quite big.